text/0000-add-ansi-escape-codes.md

```diff
@@ -0,0 +1,63 @@
+- Feature Name: add_ansi_escape_codes
+- Start Date: 2022-02-14
+- RFC PR: (leave this empty)
+- Pony Issue: (leave this empty)
+
+# Summary
+
+This RFC proposes the addition of more escape code functions to the ANSI primitive of the term package and the consolidation of the thirty-two existing color functions into four functions.
+
+# Motivation
+
+The purpose of these additional methods is to cover more of the standard ANSI escape codes within the stdlib.
+
+# Detailed design
+
+- Add escape codes for 3-bit, 4-bit, 8-bit, and 24-bit colors for the foreground and background.  
+The naming scheme of the color functions would be `{fg,bg}_{3,4,8,24}bit`.  
+The 3-bit, 4-bit, and 8-bit versions would take a U8 specifying the color.  
+The 24-bit versions would take three U8s specifying the r, g, and b values.  
+- Remove the thirty-two named color functions in favor of the 3-bit and 4-bit functions.
```

Comment:
I am not a fan of removing these functions entirely. They do overlap with the new 3-bit and 4-bit functions, but only partially. Might I propose creating `FG` and `BG` primitives or simple a `Color` primitive to move this functionality to? The usage could then appear as something like: `ASNI.fg4bit(Color.red())` or `ASNI.fg4bit(FG.red())`

Better yet (in my opinion) is leverage types a bit more to make it impossible to do something like `ASNI.fg4bit(Color.red_bg())` where an `fg` function is being used with a `bg` color.